vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt

```diff
@@ -286,8 +291,17 @@ class CreateRoomViewModel @AssistedInject constructor(@Assisted private val init
                     disableFederation = state.disableFederation
 
                     // Encryption
-                    if (state.isEncrypted) {
-                        enableEncryption()
+                    // we ignore the isEncrypted for public room as the switch is hidden in this case
+                    if (state.roomJoinRules != RoomJoinRules.PUBLIC && state.isEncrypted != null) {
```

Comment:
NIT: it might be clearer to convert this to a `when` and decouple `isEncrypted` creation from the `enableEncryption()`

```kotlin
val isEncrpyted = when {
  // public rooms use the default encrypted join rules as the switch is hidden
  state.roomJoinRules == RoomJoinRules.PUBLIC -> state.defaultEncrypted[state.roomJoinRules] ?: false
  else -> state.isEncrypted ?: false
}

if (isEncrpyted) {
  enableEncryption()
}
```

what do you think?

Reply:
`?: false` can also get out the `when` branch. We also have a nice `.orFalse` extension.
@BillCarsonFr if you copy this code, please fix the typo in `isEncrpyted` 😛

Reply:
for some reason I can't type `isEncrypted` :facepalm:

Reply:
ah ah I definitly failed to make it clearer :D